src/pages/index.js

```diff
@@ -104,88 +182,87 @@ function renderCard(item, method = "prepend") {
   cardsList[method](cardElement);
 }
 
-function handleEscClose(evt) {
-  if (evt.key === "Escape") {
-    const openModal = document.querySelector(".modal_is-opened");
-    if (openModal) {
-      closeModal(openModal);
-    }
-  }
-}
-
-function openModal(modal) {
-  modal.classList.add("modal_is-opened");
-  document.addEventListener("keydown", handleEscClose);
-}
-
-function closeModal(modal) {
-  modal.classList.remove("modal_is-opened");
-  document.removeEventListener("keydown", handleEscClose);
-}
-
-editProfileBtn.addEventListener("click", function () {
+api
+  .getAppInfo()
+  .then(([userData, cards]) => {
+    profileNameEl.textContent = userData.name;
+    profileDescriptionEl.textContent = userData.about;
+    profileAvatarEl.src = userData.avatar;
+    cards.forEach((item) => renderCard(item, "append"));
+  })
+  .catch(console.error);
+
+editProfileBtn.addEventListener("click", () => {
   editProfileNameInput.value = profileNameEl.textContent;
   editDescriptionInput.value = profileDescriptionEl.textContent;
   resetValidation(editProfileForm, validationConfig);
   openModal(editProfileModal);
 });
+newPostBtn.addEventListener("click", () => openModal(newPostModal));
+avatarModalBtn.addEventListener("click", () => openModal(avatarModal));
 
-editProfileCloseBtn.addEventListener("click", function () {
-  closeModal(editProfileModal);
-});
-
-newPostBtn.addEventListener("click", function () {
-  openModal(newPostModal);
-});
-
-newPostCloseBtn.addEventListener("click", function () {
-  closeModal(newPostModal);
-});
-
-previewModalCloseBtn.addEventListener("click", function () {
-  closeModal(previewModal);
-});
+editProfileCloseBtn.addEventListener("click", () =>
+  closeModal(editProfileModal)
+);
+newPostCloseBtn.addEventListener("click", () => closeModal(newPostModal));
+previewModalCloseBtn.addEventListener("click", () => closeModal(previewModal));
+avatarCloseBtn.addEventListener("click", () => closeModal(avatarModal));
+avatarModalDeleteBtn.addEventListener("click", () => closeModal(deleteModal));
 
-function handleEditProfileSubmit(evt) {
+editProfileForm.addEventListener("submit", (evt) => {
   evt.preventDefault();
-  profileNameEl.textContent = editProfileNameInput.value;
-  profileDescriptionEl.textContent = editDescriptionInput.value;
-  closeModal(editProfileModal);
-}
-
-editProfileForm.addEventListener("submit", handleEditProfileSubmit);
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+
+  api
+    .editUserInfo({
+      name: editProfileNameInput.value,
+      about: editDescriptionInput.value,
+    })
+    .then((data) => {
+      profileNameEl.textContent = data.name;
+      profileDescriptionEl.textContent = data.about;
+      closeModal(editProfileModal);
+    })
+    .catch(console.error)
+    .finally(() => setButtonText(submitBtn, false));
+});
 
```

Comment:
* Please, make the `delete` modal match the `mobile` design on 320px https://snipboard.io/vZ7daf.jpg